algorithms/nautilus/grammartec/mutator.cpp

```diff
@@ -0,0 +1,373 @@
+#include "fuzzuf/utils/common.hpp"
+#include "fuzzuf/utils/random.hpp"
+
+
+namespace fuzzuf::algorithm::nautilus::grammartec {
+
+/**
+ * @fn
+ * Minimize a tree with constraints by tester satisfied.
+ * @brief Subtree Minimization
+ * @param (tree) Tree
+ * @param (bits) Set of bits to be passed to tester
+ * @param (ctx) Context
+ * @param (start_index) Beginning of node ID
+ * @param (end_index) End of node ID
+ * @param (tester) Tester function to check if a tree satisfies constraints
+ * @return Returns true if minimization is complete, otherwise false
+ *
+ * @details Subtree minimization makes subtrees as short as possible
+ *          while still triggering the same paths of the original
+ *          testcase.
+ */
+bool Mutator::MinimizeTree(Tree& tree,
+                           std::unordered_set<size_t>& bits,
+                           Context& ctx,
+                           size_t start_index, size_t end_index,
+                           FTester& tester) {
+  size_t i = start_index;
+
+  /* For each nonterminal, we generate the smallest possible subtree */
+  while (i < tree.Size()) {
+    NodeID n(i);
+    NTermID nt = tree.GetRule(n, ctx).Nonterm();
+
+    /* Make sure it's the smallest */
+    if (tree.SubTreeSize(n) > ctx.GetMinLenForNT(nt)) {
+      /* We sequentially replace each node's subtree with the smallest one */
+      _scratchpad.GenerateFromNT(nt, ctx.GetMinLenForNT(nt), ctx);
+      if (std::optional<Tree> t = Mutator::TestAndConvert(
+            tree, n, _scratchpad, NodeID(0), ctx, bits, tester
+          )) {
+        /* If the new transitions are still triggered, replace the tree */
+        tree = t.value();
+      }
+    }
+
+    if (++i == end_index) {
+      return false;
+    }
+  }
+
+  return true;
+};
+
+/**
+ * @fn
+ * Minimize subtree so that it satisfies some constraints.
+ * @brief Recursive Minimization
+ * @param (tree) Tree
+ * @param (bits) Set of bits to be passed to tester
+ * @param (ctx) Context
+ * @param (start_index) Beginning of node ID
+ * @param (end_index) End of node ID
+ * @param (tester) Tester function to check if a tree satisfies constraints
+ * @return Returns true if minimization is complete, otherwise false
+ *
+ * @details Recursive minimization reduces the amount of recursions
+ *          by replacing the recursions one at a time.
+ */
+bool Mutator::MinimizeRec(Tree& tree,
+                          std::unordered_set<size_t>& bits,
+                          Context& ctx,
+                          size_t start_index, size_t end_index,
+                          FTester& tester) {
+  size_t i = start_index;
+
+  while (i < tree.Size()) {
+    NodeID n(i);
+
+    if (auto parent = Mutator::FindParentWithNT(tree, n, ctx)) {
+      if (auto t = Mutator::TestAndConvert(
+            tree, parent.value(), tree, n, ctx, bits, tester
+          )) {
+        /* If the new transitions are still triggered, replace the tree */
+        tree = t.value();
```

Comment:
`std::move(t.value())`

Reply:
Fixed in [2bf662f](https://github.com/fuzzuf/fuzzuf/pull/23/commits/2bf662f641b85151bbb79f7a6cf99a62230d868c)